sgx/demo/src/node/demo_kmip_server.c

```diff
@@ -96,11 +107,12 @@ static void demo_kmip_server_get_options(DemoServer * demo_server,
       break;
     // network Connection
     case 'p':
-      demo_server->tlsconn.port = strdup(optarg);
+      demo_server->tlsconn.conn_port = strdup(optarg);
       break;
     // Misc
     case 'h':
       demo_kmip_server_usage(argv[0]);
+      demo_kmip_server_cleanup(demo_server);
```

Comment:
I don't think it simplifies it, but I do agree that it makes more sense.

I don't believe that we can directly specify the order in which getopt() processes the command line options contained in `argv`. Commit [bc8abe9](https://github.com/NationalSecurityAgency/kmyth/pull/189/commits/bc8abe9243c9d74dd06bc11560eaaee27ed83c4f), however, changes the code to parse the options first and then process the parsed options in a specified order. While this is probably more complicated than just doing the cleanup step we can now avoid, I like that this enables combining the option validation (previously in a separate function) with the 'get_options' functionality.

In reviewing this code, I also identified a couple of related minor fixes:
- removed a non-existent 'm' option from the `getopt_long()` call
- changed the poorly named 'client-key' and 'client-cert' options to 'server-key' and 'server-cert'

Also, while this complicates things slightly, rather than the typical approach of immediate exit on error detection, an `invalid_options` boolean is used to allow first logging all detected instances of incorrect option use before calling `demo_kmip_server_error()`. Although I chose this approach, I do not have a strong preference for it.